### ISSUE TITLE
Write Color Buffers regression caused by #5115

### ISSUE THREAD

**Issue by stride21** [RPCS3/rpcs3#5180]
Since #5115 was merged, Motorstorm no longer goes ingame when using Write Color Buffers.

**Comment by weter11** [comment]
NieR Replicant [BLJM60223] also regressed. Huge performance drop, also this game never looks like on real PS3. There are some problems with something like HDR on RPCS3. 
7377
![_2018-09-26_14-25-38](https://user-images.githubusercontent.com/14630689/46079667-706e3900-c1a0-11e8-9803-a0838607355c.png)
7378
![_2018-09-26_14-17-33](https://user-images.githubusercontent.com/14630689/46079694-87ad2680-c1a0-11e8-87b8-9d25e6109265.png)
7380
![_2018-09-26_14-13-15](https://user-images.githubusercontent.com/14630689/46080082-9942fe00-c1a1-11e8-9a2a-f0179b089475.png)
The problem with the stuttering sound, when WCB is on is old.

**Comment by stride21** [comment]
@weter11 Thats an unrelated issue. If your actually seeing a performance loss report it here #5175.

**Comment by ruipin** [comment]
@weter11 Please bisect the PRs to figure out which one reduced performance. The graphics issue is not related to #5115 so unless you test and confirm it I'm going to assume the performance loss was also caused by a different PR.

@stride21 I'll try and reproduce the Motorstorm crash soon. Have you confirmed that it is my PR and not the one before it?

**Comment by weter11** [comment]
Sorry, but I thought that it is caused by #5115, but can't test. I'm on linux and don't have any binary after 7377 and before 7388 to understand which commit introduce performance drop.

**Comment by AniLeo** [comment]
https://rpcs3.net/cdn/builds/rpcs3-v0.0.5-7377-b7e010bb_linux64.AppImage
https://rpcs3.net/cdn/builds/rpcs3-v0.0.5-7380-06572c60_linux64.AppImage
https://rpcs3.net/cdn/builds/rpcs3-v0.0.5-7385-35139ebf_linux64.AppImage
https://rpcs3.net/cdn/builds/rpcs3-v0.0.5-7387-a3d44b5e_linux64.AppImage
https://rpcs3.net/cdn/builds/rpcs3-v0.0.5-7388-dab30c00_linux64.AppImage

**Comment by stride21** [comment]
@ruipin Can confirm it did. @Xcedf also reported having a similar issue with Heavy Rain.

**Comment by ruipin** [comment]
> Sorry, but I thought that it is caused by #5115, but can't test. I'm on linux and don't have any binary after 7377 and before 7388 to understand which commit introduce performance drop. Excuse me.

Sorry if I came off too strong, but you have to realize you posted a graphical/performance issue with Nier (with the graphical issue having already been fixed, and being unrelated to #5115) on a completely unrelated issue about how a different game (Motorstorm) crashes (i.e. nothing to do with graphics issues or performance) when you open it with WCB enabled since #5115.

You have to realize that if I looked into every single random report people post on threads and say they think it's #5115 I would spend all my time trying to reproduce bugs that don't exist or have nothing to do with my code.

**Comment by ruipin** [comment]
@stride21 Can I get a log file of Motorstorm when the crash occurs? I will try to take a look this weekend. I'll see if I can try out Heavy Rain as well.

**Comment by weter11** [comment]
7380: graphics distorted,but performance is ok; 7385: graphics distorted and performance drop from 34fps to 21fps. Always clean cache before tests.

**Comment by ruipin** [comment]
@weter11 Ok, that is better. Please open a separate issue about performance regressions after #5115 (build 7385) so we can track that issue better.

**Comment by stride21** [comment]
@ruipin I just wanted to let you know that Motorstorm doesn't consistently go ingame even with the previous master, using PPU fast helps but it can still crash on the main menu. Also, the first race causes a crash when selected, both the 2nd and 3rd races work.

**Comment by ruipin** [comment]
I did not manage to reproduce the issue in Motorstorm. Or to be exact, it crashes for me every time with WCB even before my PR.

I'll try Heavy Rain later...

**Comment by elhaya** [comment]
> I did not manage to reproduce the issue in Motorstorm. Or to be exact, it crashes for me every time with WCB even before my PR.
> 
> I'll try Heavy Rain later...

yeah with heavy rain it happens every time you set WCB. unfortunately the game is no longer playable because without it the gfx are all messed up.
i'll be happy to help with logs if you need me to..

**Comment by stride21** [comment]
> 
> 
> I did not manage to reproduce the issue in Motorstorm. Or to be exact, it crashes for me every time with WCB even before my PR.
> 
> I'll try Heavy Rain later...

@ruipin I can be hard to get ingame but it does go ingame with the PR before your build.
![motorstorm](https://user-images.githubusercontent.com/37915864/46249096-f8df1a80-c3e8-11e8-9027-5ff93fdb970f.JPG)

**Comment by stride21** [comment]
@ruipin #5265 fixes the race condition for Motorstorm, so you shouldn't have the issues with crashing when trying to test. The 2nd and 3rd race are the ones that regressed, the first race can go ingame even with WCB enabled with Elad's PR.

**Comment by ruipin** [comment]
I didn't look further into these crashes, but debugged driver crashes caused partially by my PR (and partially by pre-existing code). The fix however makes the Nier situation worse, so I won't submitting a PR with it until I fix the Nier issues, which are relatively complex/annoying (hopefully will finish that this weekend).

Might be worth giving my WIP fixes a try: https://github.com/ruipin/rpcs3/commits/wip-texture-cache-pr2 (https://github.com/ruipin/rpcs3/commit/62221b37cb3239c1f0af878a4e05dfe2a0889ce5)

**Comment by ruipin** [comment]
Please test #5276

**Comment by stride21** [comment]
Fixed with #5427
